src/index.ts

```diff
@@ -1,52 +1,57 @@
-import { flavorEntries } from '@catppuccin/palette'
-import type { AccentName, FlavorName, MonochromaticName } from '@catppuccin/palette'
-import type { CustomTheme } from 'daisyui'
+import type { AccentName, ColorName, FlavorName, MonochromaticName } from '@catppuccin/palette'
+import type themes from 'daisyui/theme/object'
+import type { PluginCreator } from 'tailwindcss/plugin'
+import { flavors } from '@catppuccin/palette'
+import daisyuiThemePlugin from 'daisyui/theme'
+import plugin from 'tailwindcss/plugin'
 
-const themeKeys = [
-  '--animation-btn',
-  '--animation-input',
-  '--btn-focus-scale',
-  '--rounded-badge',
-  '--rounded-box',
-  '--rounded-btn',
-  '--tab-border',
-  '--tab-radius',
-  'accent',
-  'accent-content',
-  'base-100',
-  'base-200',
-  'base-300',
-  'base-content',
-  'color-scheme',
-  'error',
-  'error-content',
-  'fontFamily',
-  'info',
-  'info-content',
-  'neutral',
-  'neutral-content',
-  'primary',
-  'primary-content',
-  'secondary',
-  'secondary-content',
-  'success',
-  'success-content',
-  'warning',
-  'warning-content',
-] as const
+interface DaisyuiOptions {
+  'isDefault'?: boolean
+  'prefersdark'?: boolean
+  'color-scheme'?: string
+  'root'?: string
+}
+
+type Theme = typeof themes[keyof typeof themes]
 
-const defaultColorOptions: ColorOptions = {
-  primary: 'lavender',
-  secondary: 'subtext1',
-  accent: 'rosewater',
-  neutral: 'overlay1',
-  success: 'green',
-  warning: 'yellow',
-  error: 'red',
-  info: 'blue',
+const defaultTheme: Theme = {
+  // Colors
+  'color-scheme': 'normal',
+  '--color-primary': 'lavender',
+  '--color-primary-content': 'text',
+  '--color-secondary': 'subtext1',
+  '--color-secondary-content': 'text',
+  '--color-accent': 'rosewater',
+  '--color-accent-content': 'text',
+  '--color-neutral': 'overlay1',
+  '--color-neutral-content': 'text',
+  '--color-success': 'green',
+  '--color-success-content': 'text',
+  '--color-warning': 'yellow',
+  '--color-warning-content': 'text',
+  '--color-error': 'red',
+  '--color-error-content': 'text',
+  '--color-info': 'blue',
+  '--color-info-content': 'text',
+  '--color-base-100': 'base',
+  '--color-base-200': 'mantle',
+  '--color-base-300': 'crust',
+  '--color-base-content': 'subtext1',
+  // Radius
+  '--radius-selector': '0.75rem',
+  '--radius-field': '0.5rem',
+  '--radius-box': '1rem',
+  // Sizes
+  '--size-field': '0.25rem',
+  '--size-selector': '0.5rem',
+  '--border': '1px',
+  // Effects
+  '--depth': '1',
+  '--noise': '0',
```

Comment:
As a colourscheme, we shouldn't be changing anything outside of the colours. Are these the defaults?

Reply:
If elements are, by default, rendered with sharp corners then I wouldn't want our colourscheme to change that automatically into something else.

I think it would make sense to link to their documentation on what extra keys can be defined alongside the theme for greater customisation?

Reply:
Ignore me, it looks like daisyUI expects these as part of a theme so I think it's okay here.
Thanks for working on supporting the latest version and sorry for the noise! @Liumingxun